xds/internal/client/filter_chain.go

```diff
@@ -166,114 +198,128 @@ func NewFilterChainManager(lis *v3listenerpb.Listener) (*FilterChainManager, err
 // internal data structures corresponding to the match criteria.
 func (fci *FilterChainManager) addFilterChains(fcs []*v3listenerpb.FilterChain) error {
 	for _, fc := range fcs {
-		// Skip filter chains with unsupported match fields/criteria.
 		fcm := fc.GetFilterChainMatch()
-		if fcm.GetDestinationPort().GetValue() != 0 ||
-			fcm.GetServerNames() != nil ||
-			(fcm.GetTransportProtocol() != "" && fcm.TransportProtocol != "raw_buffer") ||
-			fcm.GetApplicationProtocols() != nil {
+		if fcm.GetDestinationPort().GetValue() != 0 {
+			// Destination port is the first match criteria and we do not
+			// support filter chains which contains this match criteria.
```

Comment:
Should we log when we are ignoring filter chains we don't support?  (Note we ignore them for several other reasons below.)

Reply:
Done.